agent/consul/client.go

```diff
@@ -276,6 +276,7 @@ TRY:
 
 	// Move off to another server, and see if we can retry.
 	c.logger.Printf("[ERR] consul: %q RPC failed to server %s: %v", method, server.Addr, rpcErr)
+	metrics.IncrCounter([]string{"client", "rpc", "failed"}, 1)
```

Comment:
Would it be useful to add a label here for the server that the RPC failed on? It seems like it would be because the problem here is likely on the server that failed to do the RPC. What do you think @guidoiaquinti.

Maybe something like this:

`metrics.IncrCounterWithLabels([]string{"client", "rpc", "failed"}, 1, []metrics.Label{{Name: "server", Value: server.ID}})`

Not sure whether server.ID, server.Name or server.Addr.String() is most appropriate.

Reply:
`server.Name` gets my vote - it must be unique at any point in time and is human readable. Metric labels with UUIDs in are pretty unpleasant to use in practice for at-a-glance analysis.

Reply:
Sorry, I read your comment after pushing my commit. I've changed the label to `server.Name` as you suggested @banks